docs/docs/dev_docs/contracts/deploying.md

```diff
@@ -27,35 +59,101 @@ Set this only if this contract is expected to receive private notes (in such a c
 By default is set to a random value.
 Set it, if you need a deterministic contract address (same functionality as Ethereum's `CREATE2` opcode).
 
-To give you a more complete example we will deploy the `PrivateToken` contract whose artifacts are included in the `@aztec/noir-contracts` package.
+</TabItem>
+<TabItem value="js" label="Aztec.js">
+
+The `deploy()` method is generated automatically with the typescript class representing your contract. It requires the RPC to be passed (this can be the `AztecRPC` client) and any arguments to be passed to the constructor.
+
+Additionally the `.send()` method can has a few optional arguments too, which are specified in an optional object:
+- `portalContract?: EthAddress`: The L1 portal address to link the contract to. See the section on [Portals to learn more about them](./portals/main.md)
+- `contractAddressSalt?: Fr`: A salt that can be passed which is used to compute the contract address of the contract being deployed. 
+By default is set to a random value.
+Set it, if you need a deterministic contract address (same functionality as Ethereum's `CREATE2` opcode).
+
+```ts
+const tx = ExampleContract.deploy(aztecRpc).send({ 
+    portalContract: EthAddress.from("0x1234..."),
+});
+```
+
+</TabItem>
+</Tabs>
 
 ### Deploying private token contract
+To give you a more complete example we will deploy the `PrivateToken` contract whose artifacts are included in the `@aztec/noir-contracts` package.
+
 The contract has `initial_supply` and `owner` as constructor arguments.
-Because the contract sends a note to the owner specified inside the constructor, we first need to register the owner as a recipient inside the Aztec RPC with the following command:
+Because the contract sends a note to the owner specified inside the constructor, we need their public key to encrypt the note with. For this, we first need to register the owner as a recipient inside the Aztec RPC with the following command:
+
+<Tabs groupId="deployment-methods">
+<TabItem value="cli" label="Aztec CLI">
 
 ```bash
 aztec-cli register-recipient --address 0x147392a39e593189902458f4303bc6e0a39128c5a1c1612f76527a162d36d529 --public-key 0x26e193aef4f83c70651485b5526c6d01a36d763223ab24efd1f9ff91b394ac0c20ad99d0ef669dc0dde8d5f5996c63105de8e15c2c87d8260b9e6f02f72af622 --partial-address 0x200e9a6c2d2e8352012e51c6637659713d336405c29386c7c4ac56779ab54fa7
 ```
 
+</TabItem>
+<TabItem value="js" label="Aztec.js">
+
+```ts
+await aztecRpc.registerRecipient(completeAddress);
```

Comment:
Here it would probably be nice if you point to what a complete address is. Just something like adding `const completeAddress = `.